### PR TITLE
[feature] Readme and Plugin Information - feedback please

### PR DESCRIPTION
**Pull request by goetzmat**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,49 @@
...
+The Goal of the Project is to provide realtime coding assistance on a second screen to help beginners with their coding tasks.  
+This is implemented using an Android App providing real time compiler errors, tasks and coding statistics as well as providing notifications for upcoming project deadlines.
+
+## Implementation
+
+The Project consists of a Plugin for the Jetbrains IntelliJ Platform (plugin marketplace link coming soon) reporting to the connecting Android App (google play link coming soon) using WebRTC.
+This App shows the current tasks, compiler errors and coding statistics.  
+
+
+![](Screenshot.png)
```
Comment (RealWhimsy) [inline README.md:13]: Create sections for every fragment. Add a screenshot of the fragment there, and explain in detail what the fragment does exactly

File: README.md [hunk 2]
```diff
@@ -1,2 +1,49 @@
...
+![](Screenshot.png)  
+
+
+## Team
+Code Companion is developed by Fabian (@PLACEBOBRO) Lukas (@RealWhimsy) Mathias (@goetzmat) and Maximilian (@MaximilianSeewald) during the "Advanced Software Engineering" Course of Uni Regensburg.
+
+
+## Template Project
+
+A template project can be found -here-.
```
Comment (RealWhimsy) [inline README.md:22]: fix link to template project folder

File: README.md [hunk 3]
```diff
@@ -1,2 +1,49 @@
...
+
+## Task Information
+
+The "task.json" file is located inside the root of the project folder, and holds the project name, tasks and project due date.
```
Comment (RealWhimsy) [inline README.md:26]: Put this as a sub-sub-section of the **Task Fragment** overview. Make clear what the fields of the tasks.json do ("tasks" array is what is displayed in the app etc.) and explain what happens when no tasks.json is present

File: README.md [hunk 4]
```diff
@@ -1,2 +1,49 @@
...
+
+The Project consists of a Plugin for the Jetbrains IntelliJ Platform (plugin marketplace link coming soon) reporting to the connecting Android App (google play link coming soon) using WebRTC.
+This App shows the current tasks, compiler errors and coding statistics.  
+
+
+![](Screenshot.png)  
+
+
+## Team
+Code Companion is developed by Fabian (@PLACEBOBRO) Lukas (@RealWhimsy) Mathias (@goetzmat) and Maximilian (@MaximilianSeewald) during the "Advanced Software Engineering" Course of Uni Regensburg.
```
Comment (RealWhimsy) [inline README.md:17]: add commas after each person

File: README.md [hunk 5]
```diff
@@ -1,2 +1,49 @@
...
+
+## Concept
+The Goal of the Project is to provide realtime coding assistance on a second screen to help beginners with their coding tasks.  
+This is implemented using an Android App providing real time compiler errors, tasks and coding statistics as well as providing notifications for upcoming project deadlines.
+
+## Implementation
+
+The Project consists of a Plugin for the Jetbrains IntelliJ Platform (plugin marketplace link coming soon) reporting to the connecting Android App (google play link coming soon) using WebRTC.
+This App shows the current tasks, compiler errors and coding statistics.  
+
```
Comment (RealWhimsy) [inline README.md:11]: missing plugin explanations + screenshots

**Review by RealWhimsy — CHANGES_REQUESTED** [review]
Needs a lot more information.

Add sections for every main-fragment (home, tasks, error messages, profile) and explain how to use them, what they do, and when they update. These explanations should be understandable by someone who has not seen anything about our app yet, so try to be as clear as possible, even if you think something is obvious.

Provide explanations for how external files, such as tasks.json, are used in our app (due date used for notifications 7, 3 and 1 day in advance, tasks are displayed in the tasks fragment, ...) , and how the user can interact with it. As it is right now, the structure of the tasks.json is shown nicely, but not explained what the fields are used for.

Also make sure that every main-fragment is explained **in detail**. For example:
Explain that the error-messages fragment updates in real-time and shows errors of the file the user currently has open in their IDE. The user can also tap on the errors to get additional information. From that screen, the user can also launch a google query for that error, either on their phone or the connected PC. Include a screenshot of an "error detail view".
The same goes for other categories, explain what the statistics mean, that the statistics displayed are either for the current project (if user is connected) or for all projects combined (if user is not connected).

**Plugin:**
As of right now, the plugin is barely explained in the readme.
Definitely should add the following:
- How to install it
- How to use it (include **screenshot** of tool window!)
- Explain how the user can see whether they are connected or not
- How the plugin behaves upon disconnecting (e.g. user closed the phone app)

Again, a lot of this information seems obvious and self-explanatory for us, but this should be a _very detailed_ explanation for people who have never ever seen our app and plugin.

If you need any help or more information let me know, I don't have any pressing issues to work on right now.

**Comment by RealWhimsy** [comment]
Maybe also add a short **Requirements** section to the top:
IntelliJ IDEA Community version x.xx (tbd)
Smartphone running Android 6.0 (Marshmallow) or higher

**Review by RealWhimsy — APPROVED** [review]
[DOCUMENTATION]
Feature: read-me
Created by: @goetzmat 
Review by: @RealWhimsy 
Working: Yes
Documentation: Top
Clean code:  Yes
Notes: Looks solid, good job on implementing the previous feedback

**Review by MaximilianSeewald — APPROVED** [review]
[DOCUMENTATION]
Feature: read-me
Created by: @goetzmat
Review by: @MaximilianSeewald 
Working: Yes
Documentation: Top
Clean code: Yes
Notes:

**Pull request merged**